src/linclust/kmermatcher.cpp

```diff
@@ -866,24 +987,27 @@ int kmermatcher(int argc, const char **argv, const Command &command) {
     setLinearFilterDefault(&par);
     par.parseParameters(argc, argv, command, true, 0, MMseqsParameter::COMMAND_CLUSTLINEAR);
 
-    DBReader<unsigned int> seqDbr(par.db1.c_str(), par.db1Index.c_str(), par.threads,
-                                  DBReader<unsigned int>::USE_INDEX | DBReader<unsigned int>::USE_DATA);
-    seqDbr.open(DBReader<unsigned int>::NOSORT);
-    int querySeqType = seqDbr.getDbtype();
-
-    setKmerLengthAndAlphabet(par, seqDbr.getAminoAcidDBSize(), querySeqType);
-    std::vector<MMseqsParameter *> *params = command.params;
-    par.printParameters(command.cmd, argc, argv, *params);
-    Debug(Debug::INFO) << "Database size: " << seqDbr.getSize() << " type: " << seqDbr.getDbTypeName() << "\n";
-
-    if (seqDbr.getMaxSeqLen() < SHRT_MAX) {
-        kmermatcherInner<short>(par, seqDbr);
-    }
-    else {
-        kmermatcherInner<int>(par, seqDbr);
-    }
-
-    seqDbr.close();
+    float hashSeqBuffer;
+    do {
+        DBReader<unsigned int> seqDbr(par.db1.c_str(), par.db1Index.c_str(), par.threads,
+                                      DBReader<unsigned int>::USE_INDEX | DBReader<unsigned int>::USE_DATA);
+        seqDbr.open(DBReader<unsigned int>::NOSORT);
```

Comment:
You should not need re reopen the DBReader and parse parameters again

Reply:
By the reason of DBReader<T>::unmapData(), it is unavoidable to reopen DBReader. I left some comments to explain it.